### ISSUE TITLE
Use typescript namespace in JSDoc comments

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#20281]
_From @Liero on November 22, 2017 15:5_

When I install npm package, VS Code nicely shows typescript definitions in javascript file:
![typescript definitions inside js](https://user-images.githubusercontent.com/3494407/33133824-4523ffd6-cf9e-11e7-8629-3641e76e88c7.png)

I would expect, that I will be able to use these definitions in JSDoc comments:

```javascript
/**
 * compiles application javascripts using specified bundler, either browserify or watchify
 * @param {browserify.BrowserifyObject} bundler
 * @param {string} bundleFileName - output javascript file
 */
function compileJs(bundler, bundleFileName) {
```

But the parameter type is still reported as any:
![image](https://user-images.githubusercontent.com/3494407/33133976-c7ede3f0-cf9e-11e7-87ea-bd543d814bde.png)

_Copied from original issue: Microsoft/vscode#38964_

**Comment by mjbvz** [comment]
@Liero Do you import or require `broswerify` anywhere in that file?

Possibly related to #14377

**Comment by mhegazy** [comment]
looks like a duplicate of  #14377

**Comment by Liero** [comment]
@mjbvz:
> Do you import or require broswerify anywhere in that file?

 yes, I do. Otherwise I would not be able to call `browserify()`, see the first screenshot. The definitions are actually downloaded and works, but I'm not able to use it in my own JSDoc comments (see seconds screenshot)

**Comment by Liero** [comment]
@mhegazy: 
> looks like a duplicate of #14377 

Is it? I'm not sure. I'm actually importing the `browserify' module, see my previous comment

**Comment by mhegazy** [comment]
> Is it? I'm not sure. I'm actually importing the `browserify' module, see my previous comment

Well, i can not tell if you are from the screen shot above. if you are not importing it, then you need #14377 to be fixed, if you are, then you are running into https://github.com/Microsoft/TypeScript/issues/16489

**Comment by Liero** [comment]
@mhegazy:

try this pls:

```javascript
const browserify = require('browserify');
var b = browserify(); //definitions are shown in tooltips and autocomplete hints

/** @param {browserify.BrowserifyObject} bundler */
function compileJs(bundler) {
  //bundler is reported as any...
}
```

package.json:
```json
{
  "devDependencies": {
    "browserify": "^14.5.0"
  }
}
```
